app/Jobs/EncodeTrackFile.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * Pony.fm - A community for pony fan music.
+ * Copyright (C) 2015 Kelvin Zhang
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Poniverse\Ponyfm\Jobs;
+
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Log;
+use OAuth2\Exception;
+use Poniverse\Ponyfm\Jobs\Job;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Bus\SelfHandling;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Poniverse\Ponyfm\Track;
+use Poniverse\Ponyfm\TrackFile;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+class EncodeTrackFile extends Job implements SelfHandling, ShouldQueue
+{
+    use InteractsWithQueue, SerializesModels;
+    /**
+     * @var TrackFile
+     */
+    private $trackFile;
+    /**
+     * @var
+     */
+    private $isExpirable;
+
+    /**
+     * Create a new job instance.
+     * @param TrackFile $trackFile
+     * @param $isExpirable
+     */
+    public function __construct(TrackFile $trackFile, $isExpirable)
+    {
+        $this->trackFile = $trackFile;
+        $this->isExpirable = $isExpirable;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        // Start the job
+        $this->trackFile->in_progress = true;
+        $this->trackFile->update();
+
+        // Assign the source to the track's master file
```

Comment:
"Use the track's master file as the source." more precisely describes what's going on here.